File: 프로그래머스/2/87946. 피로도/피로도.java
import java.util.Arrays;
class Solution {
    
    public int answer=-1;
    
    public void dfs(int k, boolean[] visited, int[][] dungeons, int cnt){
        for(int i=0;i<dungeons.length;i++){
            if(!visited[i] && k>=dungeons[i][0]){
                visited[i]=true;
                dfs(k-dungeons[i][1],visited,dungeons,cnt+1);
                visited[i]=false;
            }
        answer = cnt>answer?cnt:answer;
        }
    }
    public int solution(int k, int[][] dungeons) {
        boolean[] visited = new boolean[dungeons.length];
        dfs(k,visited,dungeons,0);
        return answer;
    }
}

File: 프로그래머스/2/87946. 피로도/피로도.py
from itertools import permutations
def solution(k, dungeons):
    answer = -1
    for schedule in permutations(dungeons,len(dungeons)):
        current = k
        goes = 0
        for dungeon in schedule:
            needs = dungeon[0]
            cost = dungeon[1]
            if current>=dungeon[0]:
                goes+=1
                current-=cost
        answer = max([answer,goes])
    return answer